scripts/test_reports/README.md

```diff
@@ -36,6 +36,17 @@ optional arguments:
                         Milestone ID
   -P PLAN, --plan PLAN  Test plan ID
 
+  -b buginfo_file, --buginfo-file buginfo_file for sanitycheck & sanitycheckbatch runner
+                     points to a file containing bug info for the error/failure cases.
```

Comment:
Where is those bugs numbers are coming from? Can you explain a bit how is this list maintained and why do we need to maintain in a file like this? How do you know a failure actually corresponds to an existing bug?

Reply:
@nashif -- Thanks for your comments. The bug info are from the daily test result, and this PR is be used for auto daily test. 
Please refer the work process below for daily test, we want to save some time for below step 4 so that we don’t need to fill the bug info one by one in the Test Rail every day.

For the daily test, the old work process is following:
  1. Daily test script invokes the sanitycheck script to run test for all platforms
  2. Collect all test results and upload them into Test Rail for all platforms 
  3.  Investigate those failure and error test cases for the test results, submit issues in the github
  4.  Manually fill the bug info one by one in the Test Rail for those failure and error test cases
  5.  Send out the test report via mail 

After this function added into daily test, the work process is be updated following : 

  1. Daily test script invoked the sanitycheck script to run test for all platforms
  2. Collect all test results for all platforms
  3. Investigate those failure and error test cases for the test results, submit issues in the github
  4. Check whether the bug info are correct in the bug info file for those failure and error test cases . In general, the most of issues are not be fixed in next day, so the most of bug info don’t need to update, we only add/update few failure/error test cases and delete the passed cases in the bug info file. 
   5. Upload test results & bug info into Test Rail for all platforms
   6. Send out the test report via mail 

BTW, when there are more failure/error cases, it will save more time. Please also refer the current bug info file below. 

=============== bug info file  ===============

[frdm_k64f]
arch.arm.swap.common.no_optimizations.arm_thread_swap=20595
arch.arm.swap.common.fp_sharing.no_optimizations.arm_thread_swap=20595
arch.interrupt.no_optimizations.arm_interrupt=20595
kernel.timer.timer_k_define=21813

[iotdk]
portability.cmsis_rtos_v2.thread_apis=17852
portability.cmsis_rtos_v2.thread_apis_dynamic=17852
portability.cmsis_rtos_v2.thread_prio=17852
portability.cmsis_rtos_v2.thread_prio_dynamic=17852
portability.cmsis_rtos_v2.timer=17852
portability.cmsis_rtos_v2.mutex=17852
portability.cmsis_rtos_v2.mutex_lock_timeout=17852
portability.cmsis_rtos_v2.semaphore=17852
portability.cmsis_rtos_v2.mempool=17852
portability.cmsis_rtos_v2.mempool_dynamic=17852
portability.cmsis_rtos_v2.messageq=17852
portability.cmsis_rtos_v2.event_flags_no_wait_timeout=17852
portability.cmsis_rtos_v2.event_flags_signalled=17852
portability.cmsis_rtos_v2.event_flags_isr=17852
portability.cmsis_rtos_v2.thread_flags_no_wait_timeout=17852
portability.cmsis_rtos_v2.thread_flags_signalled=17852
portability.cmsis_rtos_v2.thread_flags_isr=17852
portability.cmsis_rtos_v2.thread_join=17852
portability.cmsis_rtos_v2.thread_detached=17852
portability.cmsis_rtos_v2.thread_joinable_detach=17852
portability.cmsis_rtos_v2.thread_joinable_terminate=17852

[reel_board]
net.ieee802154.fragment.fragment_sam01_dam01=19761
net.ieee802154.fragment.fragment_sam10_dam10=19761
net.ieee802154.fragment.fragment_sam00_m1_dam00=19761
net.ieee802154.fragment.fragment_sam01_m1_dam01=19761
net.ieee802154.fragment.fragment_sam10_m1_dam10=19761
net.ieee802154.fragment.fragment_ipv6_dispatch_small=19761
net.ieee802154.fragment.fragment_ipv6_dispatch_big=19761
arch.arm.irq_advanced_features.arm_dynamic_direct_interrupts=21812
arch.arm.irq_advanced_features.arm_zero_latency_irqs=21812

[mimxrt1050_evk]
usb.device.usb_dc_api_invalid=19763
usb.device.usb_dc_api_read_write=19763
usb.device.usb_deconfig=19763
usb.device.usb_disable=19763
drivers.counter.single_shot_alarm_notop=21351

[sam_e70_xplained]
logging.log_core.log_arguments=18728
logging.log_core.log_backend_runtime_filtering=18728
logging.log_core.log_overflow=18728
logging.log_core.log_panic=18728
logging.log_core.log_from_declared_module=18728
logging.log_core.log_strdup_gc=18728
logging.log_core.strdup_trimming=18728
logging.log_core.log_msg_dropped_notification=18728

[mec15xxevb_assy6853]
kernel.objects.tracing.philosophers_tracing=21756
drivers.adc.adc_sample_one_channel=21755
drivers.adc.adc_asynchronous_call=21755
drivers.adc.adc_sample_with_interval=21755
drivers.adc.adc_repeated_samplings=21755
drivers.adc.adc_invalid_request=21755
drivers.watchdog.wdt=20301
drivers.adc.adc_sample_two_channels=21755

=============== End  ===================